src/main/haskell/kore/src/Kore/Step/Substitution.hs

```diff
@@ -284,27 +285,26 @@ normalizePredicatedSubstitution
         )
     => MetadataTools level StepperAttributes
     -> Predicated level variable a
-    -> ExceptT
-          (UnificationOrSubstitutionError level variable)
-          m
-          ( Predicated level variable a
-          , UnificationProof level variable
-          )
+    -> m (Predicated level variable a)
 normalizePredicatedSubstitution
     tools@MetadataTools { symbolOrAliasSorts }
     Predicated { term, predicate, substitution }
-  = do
-    (PredicateSubstitution normalizePredicate substitution', proof)
-      <-
-        normalizeSubstitutionAfterMerge tools substitution
-    let (predicate', _) =
-            give symbolOrAliasSorts
-                (makeAndPredicate predicate normalizePredicate)
-    return
-        ( Predicated
-            { term
-            , predicate = predicate'
-            , substitution = substitution'
-            }
-        , proof
-        )
+  = give symbolOrAliasSorts $ do
+    x <- runExceptT $ normalizeSubstitutionAfterMerge tools substitution
+    return $ case x of
+      Right (PredicateSubstitution p s, _) ->
+                Predicated
```

Comment:
Please de-indent so that indentation is a multiple of 4 and use record notation as below.